### ISSUE TITLE
[Funding Request] OpenSSF Slack Upgrade to Slack Pro

### ISSUE THREAD

**Issue by JLLeitschuh** [ossf/tac#138]
To upgrade the slack instance to Slack Pro would cost $2,292.50 USD monthly or $22,794 USD yearly (there is a discount for paying yearly) for the 262 active users. It's important to note that it looks like slack doesn't bill by number of current users, only by the number of active users. This significantly decreases the actual cost.

The OpenSSF Slack channel is the lifeblood of many of the OpenSSF conversations, connections and collaboration. Not being able to retain the history of those communications beyond several weeks is limiting and prevents us from utilizing links to conversations as historical & searchable discussions regarding the rationale behind important decisions.

I understand that Slack shouldn't be where final decisions are made and recorded, but the rapid discussion format that slack offers   sometimes leads to deep interesting conversation threads. It's important that records of these can be retained.

**Comment by inferno-chromium** [comment]
I think $22K/yr seems totally reasonable given the amount of communications and member participation. I am strongly in favor of OpenSSF investing to upgrade this instance, but i don't think TAC makes the budgeting decisions for these. Can you email operations@ ?

**Comment by JLLeitschuh** [comment]
> Can you email operations@ ?

Done

**Comment by brianbehlendorf** [comment]
I think the right way to frame this question for the TAC is: do you think as a policy and technical matter, that the feature set provided by Slack Pro (which is primarily in message retention >3 months) is an essential or at least important function for the OpenSSF technical community, as opposed to treating Slack discussions as inherently ephemeral and anything important should be captured more formally through PRs or other documentation. If so, then after that we can talk about the price of this. Price is of course a part of any final decision; at current usage it's only $22k/year for 260 active users, but we should anticipate that usage growing, and yet the more it grows in usage the less flexibility we have to consider alternatives. We could also try harder to find a sympathetic ear at Slack to either offer us an unlimited plan or recognize us as a non-profit or project worth supporting. But let's separate the two steps.

**Comment by AevaOnline** [comment]
Jonathan, thanks for raising this issue.  Brian, thanks for the excellent framing.  My sense is that the current usage of Slack includes both ephemeral "coordination-style" messaging as well as deeper technical discussions that have referential value.

Some data to help inform the discussion, as I have view permissions on the instance analytics:
- our Slack instance has ~70 channels and a historical total of 97k messages sent
- As of 2/12, it had 684 _weekly active members_, and ~2,000 total accounts
- Approximately 75% of all message traffic in the last week was not in public channels

By comparison:
- lists.openssf.org has ~1,450 total accounts and 26 mailing lists
- other than "announcement" lists, most lists have less than 100 subscribers, and none have more than 200
- the TAC list is the most active, with 638 messages since its creation
- I'm unable to see the aggregate sum for all lists

Looking at this, it is clear to me that our community relies on the Slack instance more than email.

If we chose a per-user pricing model, the cost is likely higher than estimated earlier in this thread, as we have more than 260 weekly active users.

If we were to consider other alternatives, I would suggest we also consider whether other features would be valuable, such as the open-ness of a platform (Slack is, after all, not open source) or whether it supports E2E encryption of messages.

**Comment by joshuagl** [comment]
I think there's a third-way between Slack (popular, hard to track conversations over time and timezones, requires an account to even view/monitor discussion) and email list (unpopular, hard to thread in common clients, unwieldy with multiple participants).

[Python](https://discuss.python.org), [Rust](https://internals.rust-lang.org), and others are using [Discourse](https://www.discourse.org) for more of their asynchronous technical discussions with referential value.

It may be a separate issue/discussion topic (which I'd be happy to file, if so), but I would love to see more OpenSSF technical discussion move to a platform which is friendlier to asynchronous collaboration across timezones (and would be happy to contribute to efforts to establish this pattern).

**Comment by joshbressers** [comment]
I think we should try to limit this discussion to Free vs Paid Slack. If we want to discuss Slack alternatives that should be a new issue (we struggled to keep the ideas separate these discussions in the TAC call also)

**Comment by brianbehlendorf** [comment]
Further to Josh's point: there are many issues to consider when thinking about our use of real-time chat systems and possible alternatives for now, but focusing just on whether it's worth the upgrade to Slack Pro, the primary (sole?) value of which is ensuring that messages older than 90 days are still visible and searchable, the TAC discussion drifted towards whether a policy that described the value of Slack's "ephemerality" and moving anything critical or decisional to GH Issues or PRs or other systems was raised. I offered to draft a policy that could be linked to from the Slack welcome banner or other appropriate place:

https://docs.google.com/document/d/11PcWGMKG6gNE_Vc90dWkLPuaj4kPC1pwd3wj6njEGu4/edit#heading=h.7ayp9ofz2zoz

Open to comments and feedback; if/as those converge on his being useful, I'll open a PR somewhere appropriate. If that PR is accepted perhaps we can then consider this issue closable.

**Comment by hythloda** [comment]
Back to the premise that Slack conversations should be recorded, there are tools such as [linen.dev](https://www.linen.dev/)that might be helpful in this. Note these are only for public channels, but nearly all conversations we need to record should happen publicly. 
Note that this makes the public slack conversations more public, which might feel like a privacy violation.

**Comment by JasonKeirstead** [comment]
@hythloda There is also Discord and Gitter.im - both are used by a very large number of open-source projects.

**Comment by ljharb** [comment]
Oof tho, discord’s UX leaves a lot to be desired, and gitter remains awkward for anything but it’s original use case “chat for a single repo”.

**Comment by hythloda** [comment]
> @hythloda There is also Discord and Gitter.im - both are used by a very large number of open-source projects.

@JasonKeirstead I think my point was not stated well, let me try again. Discord, Gitter, Mattermost, Zulip, Rocket or any other tools require a change in community behavior. I mention linen.dev as simply an archival mechanism that does not require any behavior change. If the problem is "retain the history" then we do not need a new platform but a system that retains what we have.

**Comment by brianbehlendorf** [comment]
I'm taking the pro-active step of closing this issue. TAC issues are not the places for funding requests for infrastructural needs. As an idea, it was worth exploring, to gauge the costs and benefits. If the TAC wishes to see this happen then OpenSSF staff will work to develop options, and put it in context of other spending priorities on tooling and infrastructure for either next year's budget or a mid-year budget adjustment. But right now there does not seem to be a burning need to revisit this. Thanks for the links and ideas y'all.
